specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -0,0 +1,1049 @@
+  @doc("Unique name for metric.")
+  @visibility("read")
+  id?: string;
+
+  @doc("Azure resource id.")
+  resourceId: string;
+
+  @doc("Metric name space.")
+  metricNamespace: string;
+
+  @doc("Metric description.")
+  displayDescription?: string;
+
+  @doc("The invariant value of metric name")
+  name: string;
+
+  @doc("Metric aggregation.")
+  aggregation: string;
+
+  @doc("Metric unit.")
+  unit?: string;
+
+  @doc("Azure resource type.")
+  resourceType: string;
+}
+
+@doc("Load test run model")
+@resource("test-runs")
+model TestRun {
+  ...TestRunIdPathParameter;
+
+  @doc("Pass fail criteria for a test.")
+  passFailCriteria?: PassFailCriteria;
+
+  @doc("Auto stop criteria for a test. This will automatically stop a load test if the error percentage is high for a certain time window.")
+  @added(APIVersions.v2023_04_01_preview)
+  autoStopCriteria?: AutoStopCriteria;
+
+  @doc("""
+Secrets can be stored in an Azure Key Vault or any other secret store. If the
+secret is stored in an Azure Key Vault, the value should be the secret
+identifier and the type should be AKV_SECRET_URI. If the secret is stored
+elsewhere, the secret value should be provided directly and the type should be
+SECRET_VALUE.
+""")
+  secrets?: Record<Secret>;
+
+  @doc("Certificates metadata")
+  certificate?: CertificateMetadata;
+
+  @doc("Environment variables which are defined as a set of <name,value> pairs.")
+  environmentVariables?: Record<string>;
+
+  @doc("Error details if there is any failure in load test run")
+  @visibility("read")
+  errorDetails?: ErrorDetails[];
+
+  @doc("Test run statistics.")
+  @visibility("read")
+  testRunStatistics?: Record<TestRunStatistics>;
+
+  @doc("The load test configuration.")
+  loadTestConfiguration?: LoadTestConfiguration;
+
+  @doc("Collection of test run artifacts")
+  @visibility("read")
+  testArtifacts?: TestRunArtifacts;
+
+  @doc("Test result for pass/Fail criteria used during the test run.")
+  @visibility("read")
+  testResult?: PFTestResult;
+
+  @doc("Number of virtual users, for which test has been run.")
+  @visibility("read")
+  virtualUsers?: int32;
+
+  @maxLength(50)
+  @minLength(2)
+  @doc("Display name of a testRun.")
+  displayName?: string;
+
+  @maxLength(50)
+  @minLength(2)
+  @doc("Associated test Id.")
+  testId?: string;
+
+  @maxLength(100)
+  @doc("The test run description.")
+  description?: string;
+
+  @doc("The test run status.")
+  @visibility("read")
+  status?: Status;
+
+  @doc("The test run start DateTime(ISO 8601 literal format).")
+  @visibility("read")
+  startDateTime?: utcDateTime;
+
+  @doc("The test run end DateTime(ISO 8601 literal format).")
+  @visibility("read")
+  endDateTime?: utcDateTime;
+
+  @doc("Test run initiated time.")
+  @visibility("read")
+  executedDateTime?: utcDateTime;
+
+  @doc("Portal url.")
+  @visibility("read")
+  portalUrl?: string;
+
+  @doc("Test run duration in milliseconds.")
+  @visibility("read")
+  duration?: int32;
```

Comment:
In the GA API this is declared as `int64`. Is there some reason we need to change it here?

Reply:
Missed it Mike, changed it back to int64.